sql/snowflake/SnowflakeParser.g4

```diff
@@ -3889,7 +3900,9 @@ predicate
     | expr comparison_operator (ALL | SOME | ANY) '(' subquery ')'
     | expr NOT* BETWEEN expr AND expr
     | expr NOT* IN '(' (subquery | expr_list) ')'
```

Comment:
the keyword NOT should only be used one or zero times. So :
```g4
    | expr NOT? BETWEEN expr AND expr
    | expr NOT? IN '(' (subquery | expr_list) ')'
```

Reply:
Wasn't mine but sure fixed.